ingest/src/main/java/com/microsoft/azure/kusto/ingest/ManagedStreamingIngestClient.java

```diff
@@ -52,6 +89,17 @@ public ManagedStreamingIngestClient(ResourceManager resourceManager,
         log.info("Creating a new ManagedStreamingIngestClient from raw parts");
         queuedIngestClient = new QueuedIngestClient(resourceManager, storageClient);
         streamingIngestClient = new StreamingIngestClient(streamingClient);
+        exponentialRetryTemplate = new ExponentialRetry(ATTEMPT_COUNT);
+    }
+
+    ManagedStreamingIngestClient(ResourceManager resourceManager,
+                                        AzureStorageClient storageClient,
+                                        StreamingClient streamingClient,
+                                        ExponentialRetry retryTemplate) {
+        log.info("Creating a new ManagedStreamingIngestClient from raw parts");
+        queuedIngestClient = new QueuedIngestClient(resourceManager, storageClient);
+        streamingIngestClient = new StreamingIngestClient(streamingClient);
+        exponentialRetryTemplate = retryTemplate;
```

Comment:
Just wondering out loud - do they need this kind of configurability that they have full control over the ExponentialRetry class to pass in? Don't we know the service better than they do to to set at least the `sleepBaseSecs`? I'd expect them to pass in 1 or 2 config params, or keep `sleepBaseSecs` as non-configurable.

Reply:
It helps for the tests.
Either way that's a package-private c-tor, so the users can't access any of it

Reply:
To help with the tests, I'd make the members of the `ExponentialRetry` class `protected`.

Reply:
Why?

Reply:
So that tests can access (and change) `ExponentialRetry`'s members.

In Java, the test class is generally placed within the same package as the main class. In fact, it's usually in an identical folder, except within `test` instead of `main`, for the very reason of getting this "trick" to work.